src/Altinn.DialogportenAdapter.EventSimulator/Infrastructure/InstanceStreamer.cs

```diff
@@ -75,23 +75,25 @@ public async IAsyncEnumerable<InstanceDto> InstanceStream(
             Order.Descending => "desc",
             _ => throw new ArgumentOutOfRangeException(nameof(sortOrder), sortOrder, null)
         };
-        
+
+        if (from > to) yield break;
         var client = _clientFactory.CreateClient(Constants.MaskinportenClientDefinitionKey);
         var queryString = QueryString
             .Create("order", $"{order}:lastChanged")
             .Add("size", pageSize.ToString(CultureInfo.InvariantCulture))
             .AddIf(from.HasValue, "lastChanged", $"gt:{from?.ToUniversalTime():O}")
-            .AddIf(to.HasValue, "lastChanged", $"lt:{to?.ToUniversalTime():O}")
+            .AddIf(to.HasValue, "lastChanged", $"lte:{to?.ToUniversalTime():O}")
             .AddIf(org is not null, "org", org!)
             .AddIf(appId is not null, "org", appId!)
```

Comment:
Not part of PR, but looks like a copy-pasta error?
```suggestion
            .AddIf(appId is not null, "appId", appId!)
```

Reply:
Good catch! 🙌